ee/vellum_ee/workflows/display/workflows/base_workflow_display.py

```diff
@@ -457,15 +461,22 @@ def _serialize_workflow_trigger(self) -> Optional[JsonArray]:
         trigger_class = trigger_edges[0].trigger_class
 
         # Validate that all trigger edges use the same trigger type
-        trigger_type_mapping = get_trigger_type_mapping()
         for edge in trigger_edges:
             if edge.trigger_class != trigger_class:
                 raise ValueError(
                     f"Mixed trigger types not supported. Found {trigger_class.__name__} and "
                     f"{edge.trigger_class.__name__} in the same workflow."
                 )
 
-        # Get the trigger type from the mapping
+        # Check if this is a VellumIntegrationTrigger (factory-generated)
+        from vellum.workflows.triggers.vellum_integration import VellumIntegrationTrigger
+
+        if issubclass(trigger_class, VellumIntegrationTrigger):
+            # Special serialization for VellumIntegrationTrigger
+            return self._serialize_vellum_integration_trigger(trigger_class)
```

Comment:
[nit] will we be outlining each trigger type here in the base workflow display serialization? Would prefer that we TriggerDisplay class' own this like we do for nodes